crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -496,11 +509,11 @@ impl Account {
             self.generate_one_time_keys_if_needed();
         }
 
-        if let Some(unused) = unused_fallback_keys {
-            if !unused.contains(&DeviceKeyAlgorithm::SignedCurve25519) {
-                // Generate a new fallback key if we don't have one.
-                self.generate_fallback_key_helper();
-            }
+        // If the server supports fallback keys or if it did so in the past, shown by
+        // the existence of a fallback creation timestamp, generate a new one if
+        // we don't have one, or if the current fallback key expired.
+        if unused_fallback_keys.is_some() || self.fallback_creation_timestamp.is_some() {
```

Comment:
We don't check for the algorithm anymore? (DeviceKeyAlgorithm::SignedCurve25519)
So basically we don't care what the server says about the fallback key beeing used or not? But yet we check if the server support fallback keys?

And related to that, could we ourself detect that a fallback key was used by checking when we receive a prekey message using a fallback_key?

Reply:
> We don't check for the algorithm anymore? (DeviceKeyAlgorithm::SignedCurve25519) So basically we don't care what the server says about the fallback key being used or not? But yet we check if the server support fallback keys?

Yes, that's correct.
 
> And related to that, could we ourself detect that a fallback key was used by checking when we receive a prekey message using a fallback_key?

We can, if we extend vodozemac to give us that information, but I don't think that it makes much sense to act upon that information.